src/Components/Web.JS/src/Platform/Mono/MonoDebugger.ts

```diff
@@ -1,19 +1,32 @@
 import { WebAssemblyResourceLoader } from '../WebAssemblyResourceLoader';
 
+const BLAZOR_DEBUG_FLAG = '_blazor_debug';
+
 const currentBrowserIsChrome = (window as any).chrome
   && navigator.userAgent.indexOf('Edge') < 0; // Edge pretends to be Chrome
 
-let isDebugging = true;
+let isDebugging = sessionStorage.getItem(BLAZOR_DEBUG_FLAG);
 
 window.addEventListener('load', () => {
   const params = new URLSearchParams(window.location.search);
-  isDebugging = params.get('_blazor_debug') === 'true';
+  const isDebuggingParam = params.get(BLAZOR_DEBUG_FLAG);
+  // (1) Value from query param then (2) value in session storage then (3) default to true
+  isDebugging = isDebuggingParam !== null ? isDebuggingParam : isDebugging || 'true';
+  sessionStorage.setItem(BLAZOR_DEBUG_FLAG, isDebugging);
```

Comment:
Could we avoid writing to `sessionStorage` in the case where the flag is false? This is so we're not creating a `sessionStorage` collection in all cases in production.

Perhaps the logic could be more like:

```ts
isDebugging = (isDebuggingParam != null) // First look at querystring
  || (sessionStorage.getItem(BLAZOR_DEBUG_FLAG) == "true") // Second consider sessionStorage
  || false; // Finally, default to false

if (isDebugging) {
    sessionStorage.setItem(BLAZOR_DEBUG_FLAG, "true");
}
```

... and then it's no longer necessary to populate `isDebugging` in two different places (line 8 and line 14), and it can be statically typed as `boolean` instead of being a string.

Reply:
> Could we avoid writing to sessionStorage in the case where the flag is false?

Since we default to `true` for backwards-compatibility, we have to persist the `false` setting to session storage to avoid scenarios like this.

1. User navigates to http://localhost:5000/?_blazor_debug=false.
2. User logins in which triggers a redirect.
3. Value of isDebugging is set to true by default.

> it can be statically typed as boolean instead of being a string.

Made some tweaks to account for this. I was originally trying to avoid the excessive `== 'true'` in the code but /shrug.

Reply:
Ah, right - I hadn’t realised this back compatibility goal. 

It sounds very strange to me though. Who’s going to be putting a special param like that on their URL in development? We wouldn’t really put that on the launch URL in the template would we?

I’m not certain we do need back compatibility here, at least if it results in such a weird developer experience. 

Would it be OK to discuss the goals for this? I’ll be online in about 15 mins.